libarchive/archive_read_support_format_tar.c

```diff
@@ -1912,6 +1911,9 @@ header_pax_extension(struct archive_read *a, struct tar *tar,
 		r = pax_attribute(a, tar, entry, attr_name.s, archive_strlen(&attr_name), value_length - 1, unconsumed);
 		ext_size -= value_length - 1;
 
+		// Release the allocated attr_name (either here or before every return in this function)
+		archive_string_free(&attr_name);
```

Comment:
The `attr_name` is only used to hold onto the attribute name for the call to `pax_attribute()` just above.  So it can be released here.  We could reuse the buffer by instead freeing the string just before every return in this function.